app/src/candid/candid_parser.c

```diff
@@ -191,6 +186,16 @@ static parser_error_t readCandidNat64(parser_context_t *ctx, uint64_t *v) {
     return parser_ok;
 }
 
+parser_error_t getCandidNat64FromVec(const uint8_t *buf, uint64_t *v, uint8_t idx) {
+    *v = 0;
```

Comment:
Add check for null pointer (buf and v) and request also vector size in order to check that idx is in range